src/rest-utils.c

```diff
@@ -38,3 +38,79 @@ int coap_to_http_status(int status)
     }
 }
 
+void free_device_list(device_database_t *head)
+{
+    device_database_t *curr, *next;
+    curr = head;
+
+    while(curr != NULL)
+    {
+        next = curr->next;
+        if(curr->uuid)
+        {
+            free(curr->uuid);
+        }
+        if(curr->psk)
+        {
+            free(curr->psk);
+        }
+        if(curr->psk_id)
+        {
+            free(curr->psk_id);
+        }
+        free(curr);
+        curr = next;
+    }
+}
+
+device_database_t * alloc_device_list(size_t size)
+{
+    if(size < 1)
+    {
+        return NULL;
+    }
+    device_database_t *head, *next = NULL;
+
+    for(int i = 0; i < size; i++)
+    {
+        head = calloc(1, sizeof(device_database_t));
```

Comment:
'free_device_list' is an abort mechanism which should free all previously allocated memory, so a memory leak shouldn't happen. Also, I have avoided to use 'calloc(size, ...' because this allocates a contiguous block of memory, which is bad in terms of memory fragmentation, but I am not sure how much is this a problem.

Reply:
Well, memory fragmentation is a result of multiple alloc and free operations, however it's usually not a problem on systems with virtual memory. Anyway, I just realized that device entries must be allocated separately, because #28 will enable adding and removing each entry individually.